xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -716,26 +716,14 @@ void DX::DeviceResources::ResizeBuffers()
 
     if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG))
     {
-      ComPtr<IDXGISwapChain4> swapChain4;
-      if (SUCCEEDED(m_swapChain.As(&swapChain4)))
+      std::vector<DXGI_COLOR_SPACE_TYPE> cs = SwapChainColorSpaces();
+      std::string colorSpaces{};
+      for (auto colorSpace : cs)
       {
-        UINT colorSpaceSupport = 0;
-        std::string colorSpaces{};
-        for (UINT colorSpace = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
-             colorSpace < DXGI_COLOR_SPACE_YCBCR_STUDIO_G24_TOPLEFT_P2020; colorSpace++)
-        {
-          if (SUCCEEDED(swapChain4->CheckColorSpaceSupport(
-                  static_cast<DXGI_COLOR_SPACE_TYPE>(colorSpace), &colorSpaceSupport)) &&
-              (colorSpaceSupport & DXGI_SWAP_CHAIN_COLOR_SPACE_SUPPORT_FLAG_PRESENT) ==
-                  DXGI_SWAP_CHAIN_COLOR_SPACE_SUPPORT_FLAG_PRESENT)
-          {
-            colorSpaces.append("\n");
-            colorSpaces.append(
-                DX::DXGIColorSpaceTypeToString(static_cast<DXGI_COLOR_SPACE_TYPE>(colorSpace)));
-          }
-        }
-        CLog::LogF(LOGDEBUG, "Color spaces supported by the swap chain:{}", colorSpaces);
+        colorSpaces.append("\n");
+        colorSpaces.append(DX::DXGIColorSpaceTypeToString(colorSpace));
```

Comment:
Is possible ovoid two .append() calls with:

colorSpaces.append('\n' + DX::DXGIColorSpaceTypeToString(colorSpace));

Reply:
yes but you end up building more strings that way?